pandas/core/strings.py

```diff
@@ -1855,6 +1855,12 @@ def _wrap_result(self, result, use_codes=True,
         # before the transformation...
         if use_codes and self._is_categorical:
             result = take_1d(result, self._orig.cat.codes)
```

Comment:
I think you just need to add ``fill_value=-1`` here, you may need to pass this thru as an arg though